haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -434,11 +436,79 @@ def save(self, save_dir: Union[Path, str]):
         raise NotImplementedError(f"Saving is not implemented for {self.__class__}")
 
 
+class _BedrockEmbeddingEncoder(_BaseEmbeddingEncoder):
+    def __init__(self, retriever: "EmbeddingRetriever"):
+        boto3_import.check()
+
+        # See https://docs.aws.amazon.com/bedrock/latest/userguide/embeddings.html for more details
+        # The maximum input text is 8K tokens and the maximum output vector length is 1536
+        # Bedrock embeddings do not support batch operations
+        self.model: str = "amazon.titan-embed-text-v1"
+        self.aws_config = retriever.aws_config
+        self.client = self.initialize_boto3_client()
+
+    def initialize_boto3_client(self):
+        if self.aws_config:
+            access_key_id = self.aws_config.get("aws_access_key_id")
+            secret_access_key = self.aws_config.get("aws_secret_access_key")
+            region = self.aws_config.get("region")
+            try:
+                return boto3.client(
+                    "bedrock-runtime",
+                    aws_access_key_id=access_key_id,
+                    aws_secret_access_key=secret_access_key,
+                    region_name=region,
+                )
+            except Exception as e:
+                raise ValueError(f"AWS client error {e}")
+        else:
+            raise ValueError("Please pass boto3.client(bedrock-runtime) credentials configuration")
+
+    def embed(self, text: str) -> np.ndarray:
+        input_body = {}
+        input_body["inputText"] = text
```

Comment:
Can we also support cohere embed models? They have different request/response objects than titan.

Reply:
So you'd like to roll the cohere embedding retriever and the new bedrock embedding retriever into one function?

Reply:
No, bedrock and cohere encoders should stay separate: one encoder per high-level API. But with bedrock we cannot only use the Amazon Titan Embedding model but also Cohere Embed model. And that should be supported by BedrockEmbeddingEncoder.

Reply:
Added on November 13th, no wonder I missed it haha. I think you've got me sorted out now on all fronts 👍🏼